vocode/streaming/agent/web_socket_user_implemented_agent.py

```diff
@@ -0,0 +1,154 @@
+            await self._process()
+            restarts += 1
+            self.logger.debug(
+                "Socket Agent connection died, restarting, num_restarts: %s", restarts
+            )
+
+    def handle_incoming_socket_message(self, message: WebSocketAgentMessage) -> None:
+        self.logger.info("OUTPUT: Handling incoming message from Socket Agent: %s", message)
+
+        agent_response_message: AgentResponseMessage
+        if cast(WebSocketAgentTextMessage, message):
+            agent_response_message = TextAgentResponseMessage(text=message.data.text)
+        elif cast(WebSocketAgentTextStopMessage, message):
+            agent_response_message = TextAndStopAgentResponseMessage(text=message.data.text)
+            self.has_ended = True
+        elif cast(WebSocketAgentTextStopMessage, message):
+            agent_response_message = StopAgentResponseMessage()
+            self.has_ended = True
+        else:
+            raise Exception("Unknown Socket message type")
+
+        agent_response = OneShotAgentResponse(agent_response_message)
+
+        # Change this to new agent response format
+        interruptibile_event = InterruptibleEvent(
+            is_interruptible=self.get_agent_config().allow_agent_to_be_cut_off,
+            payload=agent_response,
+        )
+        self.logger.info("Putting interruptible agent response event in output queue")
+        self.output_queue.put_nowait(interruptibile_event)
+
+    async def _process(self) -> None:
+        extra_headers: dict[str, str] = {}
+        socket_url = self.get_agent_config().respond.url
+        self.logger.info("Connecting to web socket agent %s", socket_url)
+
+        async with websockets.connect(
+            socket_url,
+            extra_headers=extra_headers
+        ) as ws:
+            async def sender(ws: WebSocketClientProtocol) -> None:  # sends audio to websocket
+                while not self.has_ended:
+                    self.logger.info("Waiting for data from agent request queue")
+                    try:
+                        if not self.input_queue.empty():
+                            self.logger.info("Request queue is not empty")
+                        else:
+                            self.logger.info("Request queue is empty")
+
+                        transcription = await asyncio.wait_for(self.input_queue.get(), None)
+                        self.logger.info("Transcript name: %s", transcription.message)
+                        agent_request = WebSocketAgentTextMessage.from_text(transcription.message)
+                        agent_request_json = json.dumps(agent_request.to_json_dictionary())
+                        self.logger.info(f"Sending data to web socket agent: {agent_request_json}")
+                        if isinstance(agent_request, StopAgentResponseMessage):
+                            self.has_ended = True
+                        elif isinstance(agent_request, TextAndStopAgentResponseMessage):
```

Comment:
Group these into single `is instance`, then add else case for unhandled request types